File: app/models/transaction.py
from sqlalchemy import Column, Integer, Numeric, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from datetime import datetime
from app.database import Base


class Transaction(Base):
    __tablename__ = "transactions_transaction"
    
    id = Column(Integer, primary_key=True, index=True)
    account_id = Column(Integer, ForeignKey("accounts_userbankaccount.id"))
    amount = Column(Numeric(12, 2))
    balance_after_transaction = Column(Numeric(12, 2))
    transaction_type = Column(Integer)
    timestamp = Column(DateTime, default=datetime.utcnow, index=True)
    
    account = relationship("UserBankAccount", back_populates="transactions")

File: migration-to-java/src/main/java/com/banking/system/model/Transaction.java
package com.banking.system.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "transactions")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Transaction {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne
    @JoinColumn(name = "account_id", nullable = false)
    private UserBankAccount account;
    
    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal amount;
    
    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal balanceAfterTransaction;
    
    @Column(nullable = false)
    private Integer transactionType;
    
    @Column(nullable = false)
    private LocalDateTime timestamp;
    
    @PrePersist
    protected void onCreate() {
        if (timestamp == null) {
            timestamp = LocalDateTime.now();
        }
    }
}
